### PR TITLE
[12.0][FIX] mass_mailing_partner: get contextual list when importing contacts

### PR DESCRIPTION
**Pull request by chienandalu**
When importing contacts from csv or xls, the list is set by context. So
we must try to default to that context when setting the contact values.

cc @Tecnativa TT24782

### CODE REVIEW

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
You have to initialize with defaults instead:
```python
vals = self.default(self._fields.keys())
record = self.new(vals)
...
```
Also extract the same code from both create/write.

**Comment by chienandalu** [comment]
@pedrobaeza Changes done

**Review by pedrobaeza — APPROVED** [review]

File: mass_mailing_partner/models/mail_mass_mailing_contact.py [hunk 1]
```diff
@@ -40,30 +40,38 @@ def _onchange_partner_mass_mailing_partner(self):
 
     @api.model
     def create(self, vals):
-        record = self.new(vals)
+        # Ensure that defaults are loaded (e.g.: import csv or xls)
+        values_w_defaults = self.default_get(self._fields.keys())
```
Comment (yajo) [inline mass_mailing_partner/models/mail_mass_mailing_contact.py]: I'm not saying this is wrong, but it makes me wonder... why aren't the default values used by default? :thinking: 

I thought that's what "defaults" are for.

File: mass_mailing_partner/models/mail_mass_mailing_contact.py [hunk 2]
```diff
@@ -40,30 +40,38 @@ def _onchange_partner_mass_mailing_partner(self):
 
     @api.model
```
Comment (yajo) [inline mass_mailing_partner/models/mail_mass_mailing_contact.py]: I see that your optimization is done specifically for importing contacts, which usually happens with a big list of records.

Have you considered taking this opportunity to migrate this method to `@api.model_create_multi`?

File: mass_mailing_partner/models/mail_mass_mailing_contact.py [hunk 3]
```diff
@@ -40,30 +40,38 @@ def _onchange_partner_mass_mailing_partner(self):
...
+            subscription_list_ids=values_w_defaults.get(
+                'subscription_list_ids', False),
+            list_ids=values_w_defaults.get('list_ids', False)
         )
-        return super(MailMassMailingContact, self).create(new_vals)
+        return super().create(new_vals)
 
     def write(self, vals):
+        # Ensure that defaults are loaded (e.g.: import csv or xls)
+        values_w_defaults = self.default_get(self._fields.keys())
```
Comment (yajo) [inline mass_mailing_partner/models/mail_mass_mailing_contact.py]: I'm pretty sure this is a bad idea.

You'd be resetting every field to their defaults if it's not specified on a call to `write()`, and that's pretty common. :thinking:

File: mass_mailing_partner/models/mail_mass_mailing_contact.py (hunk 1 above)
Comment (chienandalu) [reply mass_mailing_partner/models/mail_mass_mailing_contact.py]: What do you think about the first approach?: https://github.com/OCA/social/commit/4f705c504b3cb855db6aafebb528dd6610c91000

File: mass_mailing_partner/models/mail_mass_mailing_contact.py (hunk 3 above)
Comment (pedrobaeza) [reply mass_mailing_partner/models/mail_mass_mailing_contact.py]: Yes, indeed this is something I haven't thought about. Anyway, why writing you need to check that context?

File: mass_mailing_partner/models/mail_mass_mailing_contact.py (hunk 1 above)
Comment (pedrobaeza) [reply mass_mailing_partner/models/mail_mass_mailing_contact.py]: The problem is that a new record doesn't apply defaults, and when converting cache to values you lose that default, as the returned dictionary contains all the fields without values, so there's no opportunity of defaults to be applied as "specifically" a value is provided. Do you get it? I still prefer this approach in create than the previous one, as it's specific for that thing, and let out any other possible default.

File: mass_mailing_partner/models/mail_mass_mailing_contact.py (hunk 3 above)
Comment (chienandalu) [reply mass_mailing_partner/models/mail_mass_mailing_contact.py]: Well, I'm guessing that with an import you could update existing records as well. But to be honest I didn't test if there was an issue in that case

Comment (pedrobaeza) [reply mass_mailing_partner/models/mail_mass_mailing_contact.py]: But doing an import you don't need to change the list. You are modifying other values (like email or similar). It's better to remove this part in the update.

File: mass_mailing_partner/models/mail_mass_mailing_contact.py (hunk 1 above)
Comment (chienandalu) [reply mass_mailing_partner/models/mail_mass_mailing_contact.py]: Ok, now I understand also :) Let's go with it

File: mass_mailing_partner/models/mail_mass_mailing_contact.py (hunk 2 above)
Comment (pedrobaeza) [reply mass_mailing_partner/models/mail_mass_mailing_contact.py]: What about this?

Comment (chienandalu) [reply mass_mailing_partner/models/mail_mass_mailing_contact.py]: Ok, done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@Yajo is it OK for you?

**Review by rafaelbn — APPROVED** [review]
:+1:

**Pull request merged**